plugins/database/src/mid_level_routines.cpp

```diff
@@ -1543,8 +1546,18 @@ int checkObjIdByTicket( const char *dataId, const char *accessLevel,
                               "update R_TICKET_MAIN set write_file_count=? where ticket_id=?",
                               icss );
                 if ( status != 0 ) {
+                    if (const auto ec = cmlExecuteNoAnswerSql("rollback", icss); ec < 0) {
+                        log_db::error("{}: Rollback error [{}].", __func__, ec);
+                    }
```

Comment:
Consider reading through the impl to make sure this is safe/necessary.

Reply:
It appears we aren't supposed to handle commit/rollback in the functions I've modified.

To highlight what I mean, `cmlCheckDataObjOnly` is called in the following db_plugin.cpp functions:
- `db_add_avu_metadata_op`
- `db_del_avu_metadata_op`
- `db_copy_avu_metadata_op`
- `db_mod_ticket_op`
- `db_reg_replica_op`
- `checkAndGetObjectId` _(called by other functions in db_plugin.cpp)_
    - `db_check_and_get_object_id_op`
    - `db_set_avu_metadata_op`
    - `db_copy_avu_metadata_op`

The majority of those functions call `_rollback()` on error which means these changes are incorrect. I think the commit and rollback are supposed to be handled in the calling function.

Assuming what I said is true, the solution to #7250 becomes more difficult because we have to review and possibly modify every function which calls `cmlCheckDataObjOnly`.

Functions calling `cmlTicketUpdateWriteBytes` must also be reviewed. Those functions are:
- `db_update_ticket_write_byte_count_op`
- `db_mod_data_obj_meta_op`

Not good at all.

Reply:
Do we anticipate this being difficult, or just very labor intensive ?

Reply:
I'm close. Tracking down why my test is failing.

Reply:
Turns out I was creating a ticket with the wrong type in the test.

It's working now.

Reply:
Now for what I've learned ...

The true cause of the bug was the GenQuery implementation.

When GenQuery is invoked with a ticket enabled on the connection, GenQuery will eventually result in a call to `cmlCheckDataObjId`. `cmlCheckDataObjId` **may** update the ticket stats in the catalog. The issue occurs because GenQuery doesn't commit or rollback the changes.